packages/elasticsearch-client/src/config/default/resources/convert-index-document.ts

```diff
@@ -49,26 +48,29 @@ export const convertIndexDocument = (
       const mapField = mappedFields[key] as string[];
       mapField.push(value.toString());
     } else {
-      mappedFields[key] = value;
+      mappedFields[key] = typeof value === 'boolean' ? value.toString() : value;
```

Comment:
I didn't really consider bools in the PoC. We should probably use a [boolean type](https://www.elastic.co/guide/en/elasticsearch/reference/current/boolean.html) in mapped field for booleans we want to filter on and set them differently here.

Not a blocker for this PR, but a TODO comment would probably be good.